internal/utils/templates/initial_schemas/14.sql

```diff
@@ -878,28 +962,31 @@ ALTER FUNCTION realtime.check_equality_op(op realtime.equality_op, type_ regtype
 CREATE FUNCTION realtime.is_visible_through_filters(columns realtime.wal_column[], filters realtime.user_defined_filter[]) RETURNS boolean
     LANGUAGE sql IMMUTABLE
     AS $$
-    /*
-    Should the record be visible (true) or filtered out (false) after *filters* are applied
-    */
-    select
-      -- Default to allowed when no filters present
-      coalesce(
-        sum(
-          realtime.check_equality_op(
-            op:=f.op,
-            type_:=col.type::regtype,
-            -- cast jsonb to text
-            val_1:=col.value #>> '{}',
-            val_2:=f.value
-          )::int
-        ) = count(1),
-        true
-      )
-    from
-      unnest(filters) f
-      join unnest(columns) col
-          on f.column_name = col.name;
-    $$;
+        /*
+        Should the record be visible (true) or filtered out (false) after *filters* are applied
+        */
+            select
+                -- Default to allowed when no filters present
+                coalesce(
+                    sum(
+                        realtime.check_equality_op(
+                            op:=f.op,
+                            type_:=coalesce(
+                                col.type_oid::regtype, -- null when wal2json version <= 2.4
+                                col.type_name::regtype
+                            ),
+                            -- cast jsonb to text
+                            val_1:=col.value #>> '{}',
+                            val_2:=f.value
+                        )::int
+                    ) = count(1),
+                    true
+                )
+            from
+                unnest(filters) f
+                join unnest(columns) col
+                    on f.column_name = col.name;
+        $$;
```

Comment:
we should probably have consistent indentations to diffs easier to read

Reply:
probably needs to be fixed in the realtime/gotrue/storage migrations, schema diffs are whitespace-sensitive for function body (it's stored as plain string)